feat(oxfmt): Use oxc_diagnostic and run format in parallel

leaysgur opened the pull request:
Part of #10573

- Following other crates, introduced `oxc_diagnostics`
  - However, `oxfmt` primarily displays only the processed paths

Boshen commented:
### Merge activity

* **Aug 29, 3:03 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(oxfmt): Use oxc_diagnostic and run format in parallel (#13378)

Part of #10573

- Following other crates, introduced `oxc_diagnostics`
  - However, `oxfmt` primarily displays only the processed paths
